app/src/main/java/com/example/rocketreserver/LoginFragment.kt

```diff
@@ -49,7 +49,7 @@ class LoginFragment : Fragment() {
                     return@launchWhenResumed
                 }
 
-                User.setToken(requireContext(), login)
+                User.setToken(requireContext(), login.token!!)
```

Comment:
It's a minor thing but I think it would be slightly nicer like this: 

```kotlin
                val token = response?.data?.login?.token
                if (token == null || response.hasErrors()) {
                    binding.submitProgressBar.visibility = View.GONE
                    binding.submit.visibility = View.VISIBLE
                    return@launchWhenResumed
                }

                User.setToken(requireContext(), token)
```

(sorry can't use a suggestion here)

Reply:
100%. Sorry I was too eager to try 3.1.0 😅